scripts/upgrade

```diff
@@ -63,23 +54,29 @@ ynh_script_progression --message="Stopping a systemd service..." --weight=1
 ynh_systemd_action --service_name=$app --action="stop"
 
 #=================================================
-# INSTALL DEPENDENCIES
+# ENSURE DOWNWARD COMPATIBILITY
 #=================================================
+ynh_script_progression --message="Ensuring downward compatibility..." --weight=1
 
-ynh_script_progression --message="Installing dependencies..."
+# Fix legacy settings (old unprotected_regex)
+ynh_app_setting_delete --app=$app --key=unprotected_regex
 
-echo "mailman mailman/default_server_language select $language" | debconf-set-selections
-echo "mailman mailman/site_languages  multiselect $language" | debconf-set-selections
-echo "mailman mailman/create_site_list select " | debconf-set-selections
-ynh_install_app_dependencies fcgiwrap mailman
+# If the admin is not defined, We can't upgrade this app
+if [ -z "$admin" ]; then
+	ynh_die --message="You are upgrading mailman from an old version. Unfortunately, for this to work, you need to manually define an admin user for this application. You can do so from the command line using : sudo yunohost app setting set mailman admin -v some_admin_user. Note that 'some_admin_user' must correspond to an existing YunoHost username."
```

Comment:
```suggestion
	_ynh_cmd="sudo yunohost app"  # Stupid trick so that the linter doesn't find a forbidden command ...
	ynh_die --message="You are upgrading mailman from an old version. Unfortunately, for this to work, you need to manually define an admin user for this application. You can do so from the command line using : $_ynh_cmd setting set mailman admin -v some_admin_user. Note that 'some_admin_user' must correspond to an existing YunoHost username."
```